Automattic/vip-go-mu-plugins
# Search: Delete inactive stale indexes

## Description
Inactive, stale indexes older than a month will be automatically deleted. This accounts for index version 1 which has no `created_time` value and uses the `activation_time` of the active index for the cut-off point.

On unsuccessful deletion, it will alert. Otherwise, it will log on Kibana.

## Changelog Description

### Plugin Updated: Enterprise Search

Automatically delete inactive indexes which are stale (a month old at least).

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Create a new index version `wp vip-search index-versions add post`
2) Activate the new index version `wp vip-search index-versions activate post <new-version-id>`
3) In testing, I changed the `$time_ago_breakpoint` to 1 minute
4) After 1 minute has passed, shell in and do the below:
`$indexable = \ElasticPress\Indexables::factory()->get( 'post' );`
`$es = new \Automattic\VIP\Search\Search();`
`$es->init();`
5) `$es->versioning_cleanup->get_stale_inactive_versions( $indexable );` and confirm that it's the old version that will be deleted
6) Delete it via `$es->versioning_cleanup->versioning_cleanup();`
7) Verify it's been deleted via `wp vip-search index-versions list post`
